ui/lib/core/addon/components/alert-banner.js

```diff
@@ -13,36 +11,14 @@ import layout from '../templates/components/alert-banner';
  * ```
  *
  * @param {String} type=null  - The banner type. This comes from the message-types helper.
- * @param {String} [secondIconType=null] - If you want a second icon to appear to the right of the title. This comes from the message-types helper.
- * @param {Object} [progressBar=null] - An object containing a value and maximum for a progress bar. Will be displayed next to the message title.
  * @param {String} [message=null] - The message to display within the banner.
+ * @param {Object} [progressBar=null] - An object containing a value and maximum for a progress bar. Will be displayed next to the message title.
+ * @param {Boolean} [showLoading=false] - Shows a loading icon to the right of the title.
  * @param {String} [title=null] - A title to show above the message. If this is not provided, there are default values for each type of alert.
- * @param {String} [bannerType=alert] - Defaults to 'alert', can be used to specify an alert banner's test selector
- *
  */
 
-export default Component.extend({
-  layout,
-  type: null,
-  message: null,
-  title: null,
-  secondIconType: null,
-  progressBar: null,
-  yieldWithoutColumn: false,
-  marginless: false,
-  classNameBindings: ['containerClass'],
-  bannerType: 'alert',
-
-  containerClass: computed('type', 'marginless', function () {
-    const base = this.marginless ? 'message message-marginless ' : 'message ';
-    return base + messageTypes([this.type]).class;
-  }),
-
-  alertType: computed('type', function () {
-    return messageTypes([this.type]);
-  }),
-
-  secondAlertType: computed('secondIconType', function () {
-    return messageTypes([this.secondIconType]);
-  }),
-});
+export default class AlertBanner extends Component {
+  get alertType() {
```

Comment:
Wow! So clean! ✨ 

I just realized that we can actually completely delete this component file since this getter is using a helper. I think that'd mean changing everything in the template from `this.alertType[keyName]` to `{{get (message-types @type) "keyName"}}`. Not necessary to do, and I think the getter probably makes the hbs template a little cleaner, just pointing out if we want to consider it 💬

Reply:
Good point. I'll argue to keep the JS file as is because we use  `this.alertType` in the hbs file as an object to gather other properties ex: `this.alertType.class` I think it becomes a little harder to read in these cases by moving the getter inside the template. Thank you for pointing this out though as it reminds me of the option to move getters inside the hbs.

Reply:
I agree!